optuna/storages/_rdb/storage.py

```diff
@@ -491,7 +491,7 @@ def _create_frozen_trial(
                     models.StudyModel.find_or_raise_by_id(study_id, session, for_update=True)
                     trial = self._get_prepared_new_trial(study_id, template_trial, session)
                     return _create_frozen_trial(trial, template_trial)
-            except sqlalchemy_exc.OperationalError as e:
+            except optuna.exceptions.StorageInternalError as e:
```

Comment:
Since `sqlalchemy_exc.OperationalError` is one of `sqlalchemy_exc.SQLAlchemyError`, the actual error is converted to `optuna.exceptions.StorageInternalError` in the `_create_scoped_session`. This change is intended.

Reply:
I see. Thank you for your modification.